fibonacci/perl/code.pl

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use feature qw(say);
+my $u;
+if(@ARGV) {
+	$u = int(shift(@ARGV));
+} else {
+	die "Please input an integer\n";
+}
+
+sub factorial {
+	my ($n, $total) = @_;
+
+	$total ||= 1;
+
+	return $total if ($n == 0);
+	@_ = ($n - 1, $n * $total);
+	goto &factorial;
+}
```

Comment:
This function needs to match the "naive" algorithm that the other implementations use. The idea is not to implement an optimally fast solution, but an equivalent solution.